retryrpc/api.go

```diff
@@ -0,0 +1,134 @@
+// Package retryrpc provides a client and server RPC model which survives
+// lost connections on either the client or the server.
+package retryrpc
+
+import (
+	"container/list"
+	"sync"
+	"time"
+)
+
+// Request is the structure sent
+type Request struct {
+	Len       uint64 // Length of this struct including len field
+	RequestID uint64 // Unique ID of this request
+	MountID   uint64 // Mount ID
+	JReq      []byte // JSON containing request
+}
+
+// Response is the structure returned
+type Response struct {
+	Len uint64 // Length of this struct including len field
+	Res []byte // JSON response - only valid if err != nil
+	Err error  // Status of request
+}
+
+type completedRequest struct {
+	completedTime time.Time
+	request       *Request
+}
+
+// Server tracks the state of the server
+type Server struct {
+	sync.Mutex
+	completedTTL      time.Duration // How long a completed request stays on queue
+	serviceMap        map[string]*func(request interface{}, response interface{})
+	connections       *list.List                   // List of outstanding connections
```

Comment:
As per my comment just following, just saying *list.List fails to reference what the Values (Elements) actually are... a comment would be more helpful if it referenced a struct corresponding to the Connections that are on the list. You certainly need such a structure if only to wrap a net.Conn and the list.Element, right?

Reply:
Not sure yet how I will manage the connections.   I think I will put it in the request that I queue on pendingRequest.   I will work on that.